internal/controlplane/identity_events.go

```diff
@@ -109,71 +109,66 @@ func HandleEvents(
 }
 
 // DeleteUser deletes a user and all their associated data from the minder database
-func DeleteUser(ctx context.Context, store db.Store, authzClient authz.Client, providerService service.GitHubProviderService,
-	userId string) (retErr error) {
+func DeleteUser(
+	ctx context.Context,
+	store db.Store,
+	authzClient authz.Client,
+	providerService service.GitHubProviderService,
+	userId string,
+) error {
 	l := zerolog.Ctx(ctx).With().
 		Str("operation", "delete").
 		Str("subject", userId).
 		Logger()
 
-	tx, err := store.BeginTransaction()
-	if err != nil {
-		return err
-	}
-	defer func() {
-		if retErr != nil {
-			if err := store.Rollback(tx); err != nil && !errors.Is(err, sql.ErrTxDone) {
-				l.Debug().Msgf("error rolling back transaction: %v", err)
-			}
-		}
-	}()
-	qtx := store.GetQuerierWithTransaction(tx)
-
-	var userDBID *int32
-
-	usr, err := qtx.GetUserBySubject(ctx, userId)
-	// If the user doesn't exist, we still want to clean up any associated data
-	if err != nil && !errors.Is(err, sql.ErrNoRows) {
-		return fmt.Errorf("error retrieving user %v", err)
-	} else if err == nil {
-		userDBID = &usr.ID
-		l = l.With().Int32("user_id", usr.ID).Logger()
-	}
-
 	// Fetching the projects for user before the deletion was made.
 	// This allows us to clean up the project from the database
 	// if there are no more role assignments for the project.
```

Comment:
I'm not sure this comment still makes sense -- we should probably update it.

Given that project IDs are UUIDs, it _seems_ reasonable to assert that we clean up the database first because any "dangling" user-to-project assignments in OpenFGA will be effectively harmless, because we should never end up reusing a project ID.  It seems worthwhile including a comment to that effect.